### PR TITLE
Spdlog integration

### PR DESCRIPTION
**Pull request by Ozaq**
Currently logging uses char* and format strings all over the place resulting in issues with filesystem::path to cstring conversion on windows.

Since there is already #310 I though Ill have a quick experiment on how a switch to spdlog might look like.

Is this change something you would like to have?

### CODE REVIEW

**Comment by Ozaq** [comment]
Hi @zeroset did you have a chance to look into this yet? Is this of interest? If yes let me know how we should proceed. Cheers!

**Comment by zeroset** [comment]
Hi, i really much like the implementation especially in combination with libfmt. I am actually not sure if we should use both libraries as a git submodule or better add a library requirement as a system library. But submodule seems to be more convenient.

I would give it a YES, but nevertheless, let's wait for an opinion from @chraibi 
Thanks!

**Comment by chraibi** [comment]
> Is this change something you would like to have?

Definitely yes.  👍 

@Ozaq:  We had this before, I think it is still important to dumb logs in a logfile *and* write out to the screen.

**Comment by chraibi** [comment]
@Ozaq  Sorry for being late, but I do love this PR. I'll check it and come back to you.

**Comment by chraibi** [comment]
Screenshot 2019-09-06 13 06 39

**Comment by Ozaq** [comment]
> To `cmake -DUSE_OPENMP=OFF ..` I get the following errors.
> Am I supposed to download something before?
> 
> Screenshot 2019-09-06 13 06 39

Woops sorry for not getting back to you, wrote an answer but never hit send... Eh yeah you would need to update the submodules but since then I reworked the PR to use system dependencies.

File: third-party/CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,7 @@ target_include_directories(visilibity PUBLIC
 # Build poly2tri
 ################################################################################
 add_library(poly2tri STATIC
+
```
Comment (zeroset) [inline third-party/CMakeLists.txt]: Important new line? ;)

**Pull request merged**